src/main/java/com/cloudant/client/org/lightcouch/internal/URIBuilder.java

```diff
@@ -38,17 +38,26 @@
     private final StringBuilder query = new StringBuilder();
     /* key=value params */
     private final Params qParams = new Params();
+    private String userInfo = "";
 
     public static URIBuilder buildUri() {
         return new URIBuilder();
     }
 
     public static URIBuilder buildUri(URI uri) {
-        URIBuilder builder = URIBuilder.buildUri().scheme(uri.getScheme())
-                .host(uri.getHost()).port(uri.getPort()).path(uri.getPath());
+        URIBuilder builder;
+        if(uri.getUserInfo() != null && !uri.getUserInfo().isEmpty()) {
```

Comment:
The reason I added getUserInfo is to support the full URI with username:password in HttpConnection's execute.  If getUserInfo is null, it won't add the proper authentication and none of the HTTP requests will work:
https://github.com/cloudant/java-cloudant/blob/856de65030fdfbff6fb731b75daa4ceb77647d8b/src/main/java/com/cloudant/http/HttpConnection.java#L190